core/bubbles/mini_workspace_bubble.ts

```diff
@@ -0,0 +1,199 @@
+      .getFlyout()
+      ?.getWorkspace()
+      ?.addChangeListener(this.updateBubbleSize.bind(this));
+    this.updateBubbleSize();
+  }
+
+  dispose() {
+    this.miniWorkspace.dispose();
+    super.dispose();
+  }
+
+  /** @internal */
+  getWorkspace(): WorkspaceSvg {
+    return this.miniWorkspace;
+  }
+
+  /** Adds a change listener to the mini workspace. */
+  addWorkspaceChangeListener(listener: (e: AbstractEvent) => void) {
+    this.miniWorkspace.addChangeListener(listener);
+  }
+
+  /**
+   * Validates the workspace options to make sure folks aren't trying to
+   * enable things the miniworkspace doesn't support.
+   */
+  private validateWorkspaceOptions(options: Options) {
+    if (options.hasCategories) {
+      throw new Error(
+        'The miniworkspace bubble does not support toolboxes with categories'
+      );
+    }
+    if (options.hasTrashcan) {
+      throw new Error('The miniworkspace bubble does not support trashcans');
+    }
+    if (
+      options.zoomOptions.controls ||
+      options.zoomOptions.wheel ||
+      options.zoomOptions.pinch
+    ) {
+      throw new Error('The miniworkspace bubble does not support zooming');
+    }
+    if (
+      options.moveOptions.scrollbars ||
+      options.moveOptions.wheel ||
+      options.moveOptions.drag
+    ) {
+      throw new Error(
+        'The miniworkspace bubble does not scrolling/moving the workspace'
+      );
+    }
+    if (options.horizontalLayout) {
+      throw new Error(
+        'The miniworkspace bubble does not support horizontal layouts'
+      );
+    }
+  }
+
+  /**
+   * Updates the size of this bubble to account for the size of the
+   * mini workspace.
+   */
+  private updateBubbleSize() {
+    if (this.miniWorkspace.isDragging()) return;
+
+    const currSize = this.getSize();
+    const newSize = this.calculateWorkspaceSize();
+    if (
+      Math.abs(currSize.width - newSize.width) < 10 &&
+      Math.abs(currSize.height - newSize.height) < 10
+    ) {
+      // Only resize if the size has noticeably changed.
+      return;
+    }
+    this.svgDialog.setAttribute('width', `${newSize.width}px`);
+    this.svgDialog.setAttribute('height', `${newSize.height}px`);
+    this.miniWorkspace.setCachedParentSvgSize(newSize.width, newSize.height);
+    if (this.miniWorkspace.RTL) {
+      this.miniWorkspace
+        .getCanvas()
+        .setAttribute('transform', `translate(${newSize.width}, 0)`);
+    }
+    this.setSize(
+      new Size(
+        newSize.width + Bubble.DOUBLE_BORDER,
+        newSize.height + Bubble.DOUBLE_BORDER
+      ),
+      this.autoLayout
+    );
+    this.miniWorkspace.resize();
+    this.miniWorkspace.recordDragTargets();
+  }
+
+  /**
+   * Calculates the size of the mini workspace for use in resizing the bubble.
+   */
+  private calculateWorkspaceSize(): Size {
+    const canvas = this.miniWorkspace.getCanvas();
+    const bbox = canvas.getBBox();
+    let width = this.miniWorkspace.RTL ? -bbox.x : bbox.width;
```

Comment:
Similarly to above: it eludes me why only `x` is relevant in RTL.  Surely a width is a width?

Reply:
Also eludes me, this was pulled over from the mutator code, but I don't think it was working 100% correctly. E.g. having a RTL bubble in a LTR workspace would break (idk why you would want to do that, but that's beside the point :P) Tested unifying the width code and it looks good! Renders correctly in LTR and RTL with a flyout and without a flyout.

Reply:
I was testing this again to make sure I caught everything, and it doesn't resize correctly as we create space between the flyout and the block. I'm just going to pull over the exact code from the existing mutator (since we know that works in common situtations). And file a bug for actually fixing / explaining this logic, and fixing the erroneous bits (e.g. RTL in LTR mode).